rdrf/rdrf/clinician_view.py

```diff
@@ -0,0 +1,260 @@
+                                NavigationFormType.CLINICIAN,
+                                None,
+                                None)
+
+    def _render_context(self, request, context):
+        context.update(csrf(request))
+        return render(request, self._get_template(), context)
+
+    def _get_objects(self, request, registry_code, patient_id):
+        self.request = request
+        self.user = request.user
+        self.message = None
+        if self.user.is_parent:
+            self.parent = ParentGuardian.objects.get(user=self.user)
+        else:
+            self.parent = None
+        self.patient_model = get_object_or_404(Patient, pk=patient_id)
+        if not security_check_user_patient(self.user, self.patient_model):
+            raise PermissionDenied
+        self.registry_model = get_object_or_404(Registry, code=registry_code)
+        if not self.registry_model.has_feature("clinician_form"):
+            raise Http404
+
+        self.patient_name = patient_name = '%s %s' % (
+            self.patient_model.given_names, self.patient_model.family_name)
+
+        try:
+            logger.debug(request.POST)
+            self.clinician_other_model = ClinicianOther.objects.get(
+                patient=self.patient_model)
+        except ClinicianOther.DoesNotExist:
+            self.clinician_other_model = None
+
+        if request.method == "POST":
+            if self.clinician_other_model is None:
+                self.clinician_form = ClinicianForm(self.registry_model,
+                                                    post_data=request.POST)
+
+            else:
+                self.clinician_form = ClinicianForm(
+                    self.registry_model,
+                    post_data=request.POST,
+                    instance=self.clinician_other_model)
+        else:
+            if self.clinician_other_model:
+                self.clinician_form = ClinicianForm(
+                    self.registry_model, instance=self.clinician_other_model)
+            else:
+                self.clinician_form = ClinicianForm(
+                    self.registry_model, initial={
+                        "patient": patient_id})
+
+        self.context_launcher = RDRFContextLauncherComponent(
+            self.user, self.registry_model, self.patient_model, "Clinician")
+
+        self.wizard = self._get_navigation_wizard()
+
+    @method_decorator(login_required)
+    def post(self, request, registry_code, patient_id):
+        self._get_objects(request, registry_code, patient_id)
+
+        if self.clinician_form.is_valid():
+            other_clinician_model = self.clinician_form.save()
+            logger.debug(
+                "saved other clinician with pk %s" %
+                other_clinician_model.pk)
+            if other_clinician_model.user:
+                other_clinician_model.patient.clinician = other_clinician_model.user
+                other_clinician_model.patient.save()
```

Comment:
should we bundle these calls to save() in a transaction so we don't end up with stuff dangling if one fails?

Reply:
Good point - been thinking of looking at transaction management across the app ( for registration also)